src/main/java/com/salesforce/dataloader/mapping/LoadMapper.java

```diff
@@ -88,8 +92,8 @@ public void verifyMappingsAreValid() throws MappingInitializationException {
         for (Map.Entry<String, String> entry : getMappingWithUnmappedColumns(false).entrySet()) {
             String sfdcName = entry.getValue();
             if(StringUtils.hasText(sfdcName)) {
-                final Field f = getClient().getField(sfdcName);
-                if (f == null)
+                final HashSet<Field> fieldSet = getClient().getField(sfdcName);
+                if (fieldSet == null | fieldSet.isEmpty())
```

Comment:
If your org doesn't have v34 yet, I don't think you can test against it, but there shouldn't be anything foundationally different.
In concept I personally don't have problem with your change. It's a matter of going more thorough testing. 
In this line, can you explain why "|" instead of "||" ? "|" does not escape the second part if the first part is true, so it might create null pointer exception. Do I get it wrong?